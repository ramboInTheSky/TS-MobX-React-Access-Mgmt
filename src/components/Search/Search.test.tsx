import React from 'react'
import { shallow } from 'enzyme'

import { Search, SearchProps } from './Search'
// const Component = (Search as any).wrappedComponent
describe('<Search />', () => {
    let props: SearchProps
    beforeEach(() => {
        props = {
            onChange: jest.fn(() => {}),
            placeholder: 'this is a placeholder',
        }
    })
    it('renders correctly', () => {
        expect(shallow(<Search {...props} />)).toMatchSnapshot()
    })
    
    it('calls onChange on clicking clear', () => {
        const component = shallow(<Search {...props} />)
        component.find('[title="Search"]').simulate('click')
        expect(props.onChange).toHaveBeenCalled()
    })
})
