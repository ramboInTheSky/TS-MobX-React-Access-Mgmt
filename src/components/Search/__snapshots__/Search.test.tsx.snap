// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<Boundary>
  <Styled(WithStyles(ForwardRef(Paper)))
    elevation={1}
  >
    <Styled(WithStyles(ForwardRef(InputBase)))
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="this is a placeholder"
      value=""
    />
    <Styled(WithStyles(ForwardRef(IconButton)))
      aria-label="Search"
      onClick={[Function]}
      title="Search"
    >
      <pure(SearchIcon) />
    </Styled(WithStyles(ForwardRef(IconButton)))>
  </Styled(WithStyles(ForwardRef(Paper)))>
</Boundary>
`;
